js/src/common/utils/ItemList.ts

```diff
@@ -1,35 +1,51 @@
-class Item {
-  content: any;
+class Item<T> {
+  content:
+    | T
+    | (T & {
+        /**
+         * Set when calling `.toArray()`
+         */
+        itemName: string;
+      });
   priority: number;
+
+  /**
+   * Set when calling `.toArray()`
+   */
   key?: number;
 
-  constructor(content: any, priority?: number) {
+  constructor(content: T, priority: number) {
```

Comment:
Why not default this? Or do we want to enforce this as recommended code style?

Reply:
The default of 0 is provided to the constructor by the `add` method on ItemList.